torch/utils/data/dataloader.py

```diff
@@ -156,6 +156,10 @@ def _pin_memory_loop(in_queue, out_queue, device_id, done_event):
         try:
             r = in_queue.get(timeout=MP_STATUS_CHECK_INTERVAL)
         except queue.Empty:
+            # If a dataloader iterator is destructed before all samples are drained, a worker process could terminate
```

Comment:
I don't think this could happen. Afterall the clean-up method in `DataLoaderIter` puts a `None` in the `worker_result_queue`.

Reply:
So under the new design, `None` is always put into `worker_result_queue` to signal a shutdown?
I will run my training to see if my hang case still happens.
Thanks for improving the DataLoader.